src/downloadlist.cpp

```diff
@@ -270,63 +251,194 @@ bool DownloadList::lessThanPredicate(const QModelIndex& left, const QModelIndex&
+      return m_manager.getFileTime(lDownloadId) <
+             m_manager.getFileTime(rDownloadId);
     } else if (left.column() == DownloadList::COL_SOURCEGAME) {
-      return m_manager.getDisplayGameName(left.row()) <
-             m_manager.getDisplayGameName(right.row());
+      return m_manager.getDisplayGameName(lDownloadId) <
+             m_manager.getDisplayGameName(rDownloadId);
     } else {
       return leftIndex < rightIndex;
     }
   } else {
     return leftIndex < rightIndex;
   }
 }
+
+Download* DownloadList::getDownload(int downloadId) {
+  auto download = std::find_if(m_downloads.begin(), m_downloads.end(),
+                      [downloadId](const Download download) {
+                        return download.downloadId == downloadId;
+                      });
+  if (download != m_downloads.end()) {
+    return download;
+  }
+
+  return nullptr;
+}
+
+int DownloadList::getDownloadIndex(int downloadId)
+{
+  auto download = std::find_if(m_downloads.begin(), m_downloads.end(),
+                               [downloadId](const Download download) {
+                                 return download.downloadId == downloadId;
+                               });
+  
+  if (download != m_downloads.end()) {
+    return download - m_downloads.begin();
+  }
+
+  return -1;
+}
+
+void DownloadList::buildDownload(Download* download) {
+  if (download->isPending) {
+    // In the case of a pending download, the downloadId will be the index
+    std::tuple<QString, int, int> nexusids = m_manager.getPendingDownload(download->downloadId);
+    download->name = tr("< game %1 mod %2 file %3 >")
+        .arg(std::get<0>(nexusids))
+        .arg(std::get<1>(nexusids))
+        .arg(std::get<2>(nexusids));
+    download->status = tr("Pending");
+    download->size   = tr("Unknown");
+  } else {
+    download->name = m_settings.interface().metaDownloads()
+                         ? m_manager.getDisplayName(download->downloadId)
+                         : m_manager.getFileName(download->downloadId);
+
+    download->size =
+        MOBase::localizedByteSize(m_manager.getFileSize(download->downloadId));
+    download->fileTime = m_manager.getFileTime(download->downloadId);
+
+    download->showInfoIncompleteWarning =
+        m_manager.isInfoIncomplete(download->downloadId);
+
+    if (!m_manager.isInfoIncomplete(download->downloadId)) {
+      const MOBase::ModRepositoryFileInfo* info =
+          m_manager.getFileInfo(download->downloadId);
+      download->modName = info->modName;
+      download->version = info->version.canonicalString();
+      download->modId   = QString("%1").arg(m_manager.getModID(download->downloadId));
+      download->game =
+          QString("%1").arg(m_manager.getDisplayGameName(download->downloadId));
+    }
+
+    switch (m_manager.getState(download->downloadId)) {
+    // STATE_DOWNLOADING handled by DownloadProgressDelegate
+    case DownloadManager::STATE_STARTED:
+      download->status = tr("Started");
```

Comment:
this is only being used for display from the enum. I've introduced a state property on the DisplayListItem that will hold the enum value and this will be the only place the the enum is translated into the display value.